### ISSUE TITLE
TypeError: __new__() missing 1 required positional argument: 'task'

### ISSUE THREAD

**Issue by levalencia** [mrdbourke/pytorch-deep-learning#377]
On video: Evaluating Our Best Models Predictions with a Confusion Matrix

You show this line:
confmat = ConfusionMatrix(num_classes=len(class_names))

But it throws above exception, task parameter is required (binary, multiclass, multilabel)

Maybe a note before the video is worth it

**Comment by vineetsharma14** [comment]
Hello There,

I have used the Confusion Matrix from TorchMetrics as below

**Create the Confusion Matrix Function**

`cm_fn = torchmetrics.ConfusionMatrix(task='multiclass', num_classes=3)`


**Pass the y_pred and y_true to the cm_fn to get the confusion matrix**

`confmat = cm_fn(y_pred, y_true)`



Hope this helps.


Thanks.

**Comment by levalencia** [comment]
Yeah I know I fixed it also on my notebook but I think it should be mentioned in a page before the video

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by alireza32000** [comment]
Hello my friend  :
I have encountered a problem with the following code, please tell me how to fix this error

from torchmetrics import ConfusionMatrix
from mlxtend.plotting import plot_confusion_matrix

# Setup confusion matrix instance
confmat = ConfusionMatrix(num_classes=len(class_names))
confmat_tensor = confmat(preds=test_preds,
                          target=test_truth)

# Plot the confusion matrix
fig, ax = plot_confusion_matrix(
     conf_mat=confmat_tensor.numpy(), # matplotlib likes working with NumPy
     class_names=class_names,
     figsize=(10, 7)
)

Related error:
TypeError: ConfusionMatrix.__new__() missing 1 required positional argument: 'task'

**Comment by otrturn** [comment]
Check out https://torchmetrics.readthedocs.io/en/stable/classification/confusion_matrix.html. The task argument has been added since the course was recorded. I ran into the same issue.
